tests/src/Unit/RiddleFeedServiceTest.php

```diff
@@ -69,6 +69,7 @@ protected function setUpConfigFactoryMock() {
     $typed_config = $this->getMock('Drupal\Core\Config\TypedConfigManagerInterface');
     $config = new Config('riddle_marketplace', $storage, $event_dispatcher, $typed_config);
     $config->set('riddle_marketplace.empty_title_prefix', 'Riddle ');
+    $config->set('riddle_marketplace.fetch_unpublished', 1);
```

Comment:
It would be nice to add tests for this.

Out of head, not checked:
```
  /**
   * Tests basic processRiddleResponse method functionality.
   *
   * @param array $riddleResponse
   *   Test case of Riddle API response.
   * @param array $expected
   *   Expected result of processRiddleResponse execution.
   *
   * @dataProvider processRiddleResponseDataProvider
   */
  public function testProcessUnpublishedRiddleResponse(array $riddleResponse, array $expected) {
    $feedService = new RiddleFeedService($this->cacheServiceMock, $this->configFactoryMock);
    $this->setProperty($feedService, 'fetchUnpublished', 0);

    $feed = $this->executeMethod(
      $feedService,
      'processRiddleResponse',
      array($riddleResponse)
    );

    unset($expected[3]);
    $this->assertEquals($expected, $feed);
  }
```